src/agent/onefuzz-task/src/tasks/fuzz/libfuzzer/common.rs

```diff
@@ -291,28 +328,99 @@ where
             }
         }
 
-        for file in &files {
+        // name the dumpfile after the crash file (if one)
+        // otherwise don't rename it
+        let dump_file_name: Option<OsString> = if files.len() == 1 {
+            files
+                .first()
+                .and_then(|f| f.file_name().map(OsStr::to_os_string))
+        } else {
+            None
+        };
+
+        // move crashing inputs to output directory
+        for file in files {
             if let Some(filename) = file.file_name() {
                 let dest = self.config.crashes.local_path.join(filename);
-                if let Err(e) = tokio::fs::rename(file.clone(), dest.clone()).await {
+                if let Err(e) = tokio::fs::rename(file, dest.clone()).await {
                     if !dest.exists() {
                         bail!(e)
                     }
                 }
             }
         }
 
+        // check for core dumps on Linux:
+        // note that collecting the dumps must be enabled by the template
+        #[cfg(target_os = "linux")]
+        if let Some(pid) = pid {
+            // expect crash dump to exist in CWD
+            let filename = format!("core.{pid}");
+            let dest_filename = dump_file_name.as_deref().unwrap_or(OsStr::new(&filename));
+            let dest_path = self.config.crashdumps.local_path.join(dest_filename);
+            match tokio::fs::rename(&filename, &dest_path).await {
+                Ok(()) => {
+                    info!(
+                        "moved crash dump {} to output directory: {}",
+                        filename,
+                        dest_path.display()
+                    );
+                }
+                Err(e) => {
+                    if e.kind() == std::io::ErrorKind::NotFound {
+                        // okay, no crash dump found
+                        warn!("no crash dump found with name: {}", filename);
+                    } else {
+                        return Err(e).context("moving crash dump to output directory");
+                    }
+                }
+            }
+        } else {
+            warn!("no PID found for libfuzzer process");
+        }
+
+        // check for crash dumps on Windows:
+        #[cfg(target_os = "windows")]
+        {
+            let dumpfile_extension = Some(std::ffi::OsStr::new("dmp"));
+
+            let mut working_dir = tokio::fs::read_dir(".").await?;
```

Comment:
Is this `.` because we're running the libfuzzer target from `.` (because we don't specify a working directory when running the command)?

Reply:
Also, we should log a warning like we do in the linux impl if we don't find any dump files?

Reply:
> Is this `.` because we're running the libfuzzer target from `.` (because we don't specify a working directory when running the command)?

Yes, that's right. The dump will appear in the current working directory.

Reply:
Added the warning, reduced both of the log messages to info.